src/librustc_typeck/check/op.rs

```diff
@@ -186,6 +186,24 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                             "binary operation `{}` cannot be applied to type `{}`",
                             op.node.as_str(),
                             lhs_ty);
+
+                        if let TypeVariants::TyRef(_, ref ty_mut) = lhs_ty.sty {
+                            match self.lookup_op_method(expr, ty_mut.ty, vec![rhs_ty_var],
+                                        token::intern(name), trait_def_id,
+                                        lhs_expr) {
+                                Ok(_) => {
```

Comment:
Oh, not sure why I missed this before, but this could also be an `if_let` or even just `if self.lookup_op_method(..).is_ok()`